app/components/model-renderer.jsx

```diff
@@ -0,0 +1,144 @@
+              </Markdown>
+              <button className="standard-button" onClick={reject}>Close</button>
+            </div>
+          )
+      );
+      /* eslint-enable no-console */
+      return;
+    }
+    // component has mounted, initialise the regl canvas
+    this.model = new Model(this.canvas, this.props.subject.metadata);
+  }
+  componentWillReceiveProps() {
+    // component has updated with new props, set the new render functions
+    // check if a model has been assigned
+    if (this.model !== null) {
+      // if so, update the render function list to the new annotation
+      window.requestAnimationFrame(this.getDifference);
+    }
+  }
+
+  // don't want to re-render whenever props update
+  shouldComponentUpdate() {
+    return this.state.imgUrl !== Object.values(this.props.subject.locations[0])[0];
+  }
+  componentWillUpdate() {
+    const im0 = this.props.subject.locations[0];
+    if (this.props.subject.locations.length < 2) {
+      this.props.subject.locations[1] = Object.assign({}, im0);
+    }
+    this.model.kill();
+    const metadata = this.props.subject.metadata;
+    this.model = new Model(this.canvas, metadata);
+    this.setState({ imgUrl: Object.values(im0)[0] });
+  }
+  componentDidUpdate() {
+    const metadata = this.props.subject.metadata;
+    this.model = new Model(this.canvas, metadata);
+  }
+  componentWillUnmount() {
+    this.model.kill();
+    this.model = null;
+  }
+  getDifference() {
+    // update the model from the annotation
+    this.model.update(this.props.classification.annotations);
+    window.requestAnimationFrame(() => {
+      // scoring function is provided in the difference calculator
+      this.props.onRender(this.model.getScore());
+      const imOutType = Object.keys(this.props.subject.locations[1])[0];
+      this.props.subject.locations[1][imOutType] = this.canvas.toDataURL(imOutType);
+    });
+  }
+  setTexture() {
+    this.model.setBaseTexture(this.im);
+    window.requestAnimationFrame(
+      () => this.model.update(this.props.classification.annotations)
+    );
+  }
+  render() {
+    const metadata = this.props.subject.metadata;
+    const size = (typeof metadata.imageSize) !== 'undefined' ? metadata.imageSize : [512, 512];
+    return (
+      <div>
+        <canvas
+          width={size[0]}
+          height={size[1]}
+          data-style={{ backgroundColor: '#000' }}
+          ref={(r) => { this.canvas = r; }}
+          hidden={true}
+        />
+        <img
+          src={`${this.state.imgUrl}?_${new Date().getTime()}`}
+          onLoad={this.setTexture}
+          ref={(r) => { this.im = r; }}
+          alt="subject to classify"
+          crossOrigin="anonymous"
+          hidden={true}
+        />
+      </div>
+    );
+  }
+}
+/* eslint-disable react/forbid-prop-types */
+ModelRenderer.propTypes = {
+  classification: React.PropTypes.object,
+  subject: React.PropTypes.object,
```

Comment:
Use `React.PropTypes.shape({})` here and define the properties of `subject` that are used in this component.